provisioning/service/src/ProvisioningServiceClient.cs

```diff
@@ -46,9 +48,13 @@ public ProvisioningServiceClient(string connectionString, ProvisioningServiceCli
                 _provisioningConnectionString,
                 clientOptions);
 
-            IndividualEnrollments = new IndividualEnrollmentsClient(_provisioningConnectionString, _contractApiHttp);
-            EnrollmentGroups = new EnrollmentGroupsClient(_provisioningConnectionString, _contractApiHttp);
-            DeviceRegistrationStates = new DeviceRegistrationStatesClient(_provisioningConnectionString, _contractApiHttp);
+            _retryPolicy = clientOptions.RetryPolicy ?? new ProvisioningServiceNoRetry();
```

Comment:
Why no retry by default?

Reply:
No, the [default](https://github.com/Azure/azure-iot-sdk-csharp/blob/brycewang/retry-dps-service/provisioning/service/src/ProvisioningServiceClientOptions.cs#L39) retry policy would be exponential backoff.

`ProvisioningServiceNoRetry` would be used when RetryPolicy is set to `null`.